pkg/config/config.go

```diff
@@ -364,6 +386,64 @@ func defaultConfigTemplate(channelConfig Channel) (*cb.ConfigGroup, error) {
 	return channelGroup, nil
 }
 
+// newSystemChannelBlock generates a genesis block by the config group and system channel ID
+// the block num is always zero
+func newSystemChannelBlock(cg *cb.ConfigGroup, channelID string) (*cb.Block, error) {
+	payloadChannelHeader := channelHeader(cb.HeaderType_CONFIG, msgVersion, channelID, epoch)
+	nonce, err := newNonce()
+	if err != nil {
+		return nil, fmt.Errorf("try to get nonce: %v", err)
+	}
+	payloadSignatureHeader := &cb.SignatureHeader{Creator: nil, Nonce: nonce}
+	payloadChannelHeader.TxId = computeTxID(payloadSignatureHeader.Nonce, payloadSignatureHeader.Creator)
```

Comment:
given that you generate `TxId` from fields of `SignatureHeader` doesn't it make sense to pass pointer to the struct itself rather than fields? also it's seems that creator is always a nil.

Reply:
Function computeTxID should be a general function. If the input parameter is *SignatureHeader, the versatility of this function will be greatly reduced. 
The design of function newBlock  is also based on the same logic, which constructs a block with no data and no metadata.
In addition, these two functions are referenced in the “protoutil/blockutils.go” file.
These are some of my thoughts, I don't know if they are suitable.